crypto/asn1/a_gentm.c

```diff
@@ -22,7 +22,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
     static const int min[9] = { 0, 0, 1, 1, 0, 0, 0, 0, 0 };
     static const int max[9] = { 99, 99, 12, 31, 23, 59, 59, 12, 59 };
```

Comment:
The leap second can be inserted at the end of any month (preferably in June and December). I was trying to spot the fact that 20161231235960Z is not considered valid with the current code, while 20160231120000Z is considered valid. And there can be only one leap second inserted (or removed). "2017.6.1 23:59:60" is a valid date (wrt ISO8601), it just does not exist, that's different.